transcom/mymove
# MB-15424 Change routing for a PPM when the postal code or origin duty station is in a different GBLOC

## [Jira ticket](https://dp3.atlassian.net/browse/MB-15424?atlOrigin=eyJpIjoiYjllNDcwYjYzYjc1NDhhNGE4YmQ3YjdhOTA0NDQxOTAiLCJwIjoiaiJ9) for this change

## Summary

A services counselor in a transportation office that provides Services Counseling should see all moves with PPMs that have selected a closeout office that matches the GBLOC of their transportation office that is ready for payment approval. This wasn't the case when the postal code or origin duty station is in a different GBLOC for AirF and Army, so I've made modifications so it should now route correctly.

To test:

1. Create a new move with a customer in either the Army or Air Force branch.
2. Create a PPM shipment and select a closeout office that has a GBLOC different from where their pickup postal code maps to. 
3. Approve the PPM as the services counselor based on the user's origin duty location
4. As the customer, submit any required PPM closeout documentation.
5. Log in as a services counselor that has the same GBLOC as the closeout office provided.
6. The move should now appear in the PPM closeout queue.

I’ve added a screenshot of one example I tried that, after this new code, now successfully routes to closeout queue/tab. It would be great for someone to test something different than what I chose, just note that you’ll want to pick a location (‘city’ in the query results) that provides services counseling:
```
SELECT * FROM public.addresses 
JOIN public.duty_locations ON duty_locations.address_id = addresses.id
WHERE provides_services_counseling = true
```

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps



1. Access the
2. Login as a
3.

## Verification Steps for Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.



### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
<img width="1187" alt="should route to ppm closeout tab now" src="https://user-images.githubusercontent.com/108361430/226730503-506ac75d-b353-4e74-9d6b-3708dc1c66c8.png">

![test_ppm_move_for_ticket_scenario](https://user-images.githubusercontent.com/108361430/226730618-de46dcad-2b36-4de8-9645-f4591fe57d9b.png)
